File: basics/30-factors/Solution.java
import java.util.*;

class Solution {
    public List<Integer> factors(int n) {
        List<Integer> result = new ArrayList<>();
        for (int i = 1; i * i <= n; i++) {
            if (n % i == 0) {
                result.add(i);
                if (i != n / i) result.add(n / i);
            }
        }
        Collections.sort(result);
        return result;
    }
}

File: basics/30-factors/Solution.cpp
#include <vector>
#include <algorithm>

std::vector<int> factors(int n) {
    std::vector<int> result;
    for (int i = 1; i * i <= n; ++i) {
        if (n % i == 0) {
            result.push_back(i);
            if (i != n / i) result.push_back(n / i);
        }
    }
    std::sort(result.begin(), result.end());
    return result;
}
